cmake/AutodetectSSE.cmake

```diff
@@ -0,0 +1,14 @@
+macro(autodetect_sse)
+    include(CheckCXXSourceRuns)
+    set(CMAKE_REQUIRED_FLAGS "")
+    if (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG)
+        set(CMAKE_REQUIRED_FLAGS "-msse4.1")
+    endif (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG)
+
+    check_cxx_source_runs("
```

Comment:
This may be more of a stretch goal, but won't "check runs" prevent cross compilation?

Reply:
These things only check if it compiles. So that should just work as well for cross compilation. Pretty sure not everything in this branch cross-compiles currently, but this is one that should "just work" as far as I know. Any reason to think otherwise that we can already take into account?

Reply:
According to the manual, `check_cxx_source_runs` explicitly tests that the compiled executable also executes on the build platform. If it just needs to compile then `check_cxx_source_compiles` should be the right one, although I'm not sure if that also confirms it can link to a valid executable.

Reply:
You are absolutely right, and this is a bug. It should be check_XXX_compiles. As indeed, running is only native :D And we just want to check if it compiles .. and it does link, so we are fine there :) Nice catch :D If you find any more, let me know!